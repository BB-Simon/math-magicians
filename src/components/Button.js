import PropTypes from 'prop-types';

const Button = ({ className, title }) => (
  <button type="button" className={className}>{ title }</button>
);

Button.defaultProps = {
  title: '',
};

Button.propTypes = {
  className: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default Button;
