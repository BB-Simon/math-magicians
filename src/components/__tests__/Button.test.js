import TestRenderer from 'react-test-renderer';
import Button from '../Button';

describe('Jest Snapshot testing suite', () => {
    it('renders correctly', () => {
    const title = "button";
    const click = () => {};
    const tree = TestRenderer.
    create(<Button onClick={click} type="button" className="button">{title}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
