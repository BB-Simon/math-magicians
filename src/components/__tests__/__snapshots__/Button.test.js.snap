// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
/>
`;
